pkg/secrethub/secret.go

```diff
@@ -8,12 +8,31 @@ import (
 
 // SecretService handles operations on secrets from SecretHub.
 type SecretService interface {
-	// Delete removes the secret at the given path.
-	Delete(path string) error
+	// Write encrypts and writes any secret data to SecretHub, always creating
+	// a new secret version for the written data. This ensures secret data is
+	// never overwritten.
+	//
+	// To ensure forward secrecy, a new secret key is used whenever the previously
+	// used key has been flagged.
+	//
+	// Write accepts any non-empty byte data that is within the size limit of MaxSecretSize.
+	// Note that data is encrypted as is. Sanitizing data is the responsibility of the
+	// function caller.
+	Write(path string, data []byte) (*api.SecretVersion, error)
+	// Read is an alias of `Versions().GetWithData` and gets a secret version, with sensitive data decrypted.
+	Read(path string) (*api.SecretVersion, error)
+	// ReadString is a convenience function to get the secret data as a string.
+	//
+	// See .Versions() for more elaborate use.
+	ReadString(path string) (string, error)
 	// Exists returns whether a secret exists on the given path.
 	Exists(path string) (bool, error)
 	// Get retrieves a Secret.
 	Get(path string) (*api.Secret, error)
+	// Versions returns a SecretVersionService.
+	Versions() SecretVersionService
```

Comment:
What made you put `Versions` here? In other places we have the 'sub services' at the end I noticed.

Reply:
I thought it belonged there since it has to do mostly with reading secrets, however it  would indeed be more consistent to have it at the end.